### PR TITLE
Feature/semi packed problem

### PR DESCRIPTION
**Pull request by etragas-fathom**
Support packed problems
https://app.asana.com/0/search/1146116926638345/1145637275439494

### CODE REVIEW

File: tensor2tensor/utils/decoding.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
 from tensor2tensor.data_generators import text_encoder
 from tensor2tensor.data_generators import text_problems
 from tensor2tensor.utils import registry
+
+from fathomairflow.dags.dag_management.common import PACKED_TO_PREDICTION_PROBLEM
```
Comment (cbockman) [inline tensor2tensor/utils/decoding.py]: hmm, dag management doesn't really seem where this list should be living?

File: tensor2tensor/utils/decoding.py [hunk 2]
```diff
@@ -202,7 +204,9 @@ def decode_from_dataset(estimator,
   ##############
 
   # Get the predictions as an iterable
-  predictions = estimator.predict(infer_input_fn)
+  packed_problem = problem_name in PACKED_TO_PREDICTION_PROBLEM.values()
+  predictions = estimator.predict(
```
Comment (cbockman) [inline tensor2tensor/utils/decoding.py]: this is a little opaque as to why things are handled differently...can we get a note?

File: tensor2tensor/utils/decoding.py (hunk 1 above)
Comment (etragas-fathom) [reply tensor2tensor/utils/decoding.py]: So we do some validation of our problems at dag construction time, as per https://github.com/medicode/diseaseTools/blob/develop/fathomairflow/dags/dag_management/dag_constructors/nlp_trainer.py#L244

But our dag_management code is separated from the rest of our code base, to keep airflow from getting bogged down by slow imports. So it has to be somewhere in dag_management.

File: tensor2tensor/utils/decoding.py (hunk 2 above)
Comment (etragas-fathom) [reply tensor2tensor/utils/decoding.py]: Added

**Pull request merged**